pkg/ccl/changefeedccl/kvevent/chunked_event_queue.go

```diff
@@ -117,7 +119,7 @@ func (bec *bufferEventChunk) pop() (e Event, ok bool, consumedAll bool) {
 
 	e = bec.events[bec.head]
 	bec.head++
-	return e, true, false
+	return e, true, bec.head == bec.tail
```

Comment:
Can you clarify what these changes fix? I'm not too sure about them.
For example, if `bec.head == bec.tail` is true, `dequeue` will free the chunk. If `bufferEventChunkArrSize < bufferEventChunkArrSize`, we will end up skipping values right?

Reply:
I think returning `bec.head == bufferEventChunkArrSize` should do the trick.

Reply:
Good idea 👍